### ISSUE TITLE
timeout_after parameter not working as expected

### ISSUE THREAD

**Issue by seif-fmfx** [OctopusDeploy/await-task-action#108]
**Describe the bug**
timeout_after parameter not working as expected

**To Reproduce**
set the timeout_after parameter

**Expected behavior**
value is respected

**Desktop (please complete the following information):**
 - OS: windows
 - Browser edge
 - Version irrelecant

**Smartphone (please complete the following information):**

**Additional context**
Attemptede a fix here https://github.com/OctopusDeploy/await-task-action/pull/107

**Comment by seif-fmfx** [comment]
For anyone else that comes here,  we changed the parameter we are setting to `timeout` instead of `timeout_after` which gives you a warning from gha but it works (i.e. the timeout is respected) .

**Comment by mfalconi-perle** [comment]
Is this what is causing new seen issues, wherein a failure is received before the task can actually complete on the octopus deploy service?

**Comment by seif-fmfx** [comment]
@mfalconi-perle sounds like it. for us it manifested as:

`waiting for task ServerTasks-98292. Status: Executing.
Error: 🐙 Could not determine Octopus Deploy task state`

but the deployment in octopus was still running

**Comment by mfalconi-perle** [comment]
> @mfalconi-perle sounds like it. for us it manifested as:
> 
> `waiting for task ServerTasks-98292. Status: Executing. Error: 🐙 Could not determine Octopus Deploy task state`
> 
> but the deployment in octopus was still running

Yeah this is exactly what I am getting on my actions as well. Thanks, glad to know this was reported. We have been confused seeing our changes and seeing the deployments running but always getting failures on the actions report

**Comment by cmyllis** [comment]
@hnrkndrssn Fix done here seems wrong: https://github.com/OctopusDeploy/await-task-action/issues/22
Can you re-open it?

**Comment by seif-fmfx** [comment]
@hnrkndrssn  @cmyllis that fix is correct, but highlighted another problem which this issue is about and that is that task is reading the input from a parameter called `timeout` when it should be reading `timeout_after`

**Comment by bnelsonjha** [comment]
@seif-fmfx Thanks for doing on this! Timeout is working for me.  I am using the latest version.  Either the code should be fixed or the front page documentation should be fixed.

**Comment by mfalconi-perle** [comment]
Was there a new release in regards to this issue? Seeing a new Actions warning pop up as of today

Unexpected input(s) 'timeout', valid inputs are ['server_task_id', 'polling_interval', 'timeout_after', 'hide_progress', 'server', 'api_key', 'space']

Whereas before timeout was working where timeout_after was not has this been fixed then?

**Comment by seif-fmfx** [comment]
@mfalconi-perle you get that warning because the action says the the field is called [timeout_after](https://github.com/OctopusDeploy/await-task-action/blob/e4b2d3f7ab6f67be509682cd618aaa2660a02977/action.yml#L15) but you are sending `timeout` which doesnt match what the action is configured to read.  Nothing has changed so you need to use `timeout`  to [control ](https://github.com/OctopusDeploy/await-task-action/blob/e4b2d3f7ab6f67be509682cd618aaa2660a02977/src/input-parameters.ts#L46) it and ignore that warning.

**Comment by iarslanch** [comment]
Error: Error: Input required and not supplied: server_task_id

using this  
 - name: Await task in Octopus Deploy 🐙
        uses: OctopusDeploy/await-task-action@v3
        env:
          OCTOPUS_API_KEY: ${{ secrets.OCTOPUS_API_KEY }}
          OCTOPUS_URL: ${{ secrets.OCTOPUS_SERVER }}
          OCTOPUS_SPACE: ${{ secrets.OCTOPUS_SPACE }}
        with:
          server_task_id: ${{ steps.deploy.outputs.server_task_id }}
          timeout_after: 600
          hide_progress: false
